### PR TITLE
Add all demos to the repo

### PR DESCRIPTION
**Pull request by rawalexe**
Adds all the demos using common-io-basic as standard 

Contributors: @cookpate @saidrhs

### CODE REVIEW

**Comment by DBees** [comment]
@rawalexe thanks for the major contribution. I note that many of the files don't have headers - could please add a header to the ones that AWS has created with the copyright and license information?

**Comment by rawalexe** [comment]
hello @DBees,
I have updated the files to include the AWS apache license information, please let me know if you have any other concerns.

File: common-io/tests/unity_main.c [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	/* Start the tasks and timer running. */
+	vTaskStartScheduler();
+
+	/* If all is well, the scheduler will now be running, and the following
+	 line will never be reached.  If the following line does execute, then
+	 there was insufficient FreeRTOS heap memory available for the Idle and/or
+	 timer tasks to be created.  See the memory management section on the
+	 FreeRTOS web site for more details on the FreeRTOS heap
+	 http://www.freertos.org/a00111.html. */
+
```
Comment (n9wxu) [inline common-io/tests/unity_main.c:61]: should make this line an assert so the lack of memory will cause an immediate crash at a specific point.

**Comment by DBees** [comment]
@rawalexe thanks but many files still don't have headers. Can we setup a quick call on Wednesday to chat? Are you on mattermost? if not contact me on duncan@openhwgroup.org

**Comment by DBees** [comment]
@MikeOpenHWGroup and I have reviewed this together. Remaining files without headers are mainly .xml files. I will merge this and put a statement in the LICENSE.MD file. We may need to revisit this. thanks @rawalexe

**Pull request merged**